### PR TITLE
#345 - Require `of` in protocol action rules

### PR DESCRIPTION
**Pull request by diehuxx**
#345

Tweaking JSON schema and making tests hopefully a little more robust in order to enforce presence of `of` when `who` === `author` or `recipient`.

### CODE REVIEW

File: json-schemas/protocol-rule-set.json [hunk 1]
```diff
@@ -9,31 +9,57 @@
...
+            "required": [
+              "who",
+              "can"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "who": {
+                "type": "string",
+                "enum": [
+                  "anyone"
```
Comment (thehenrytsai) [inline json-schemas/protocol-rule-set.json:23]: Consider adding a test: 'should throw if `of` is specified for `anyone` rule'

File: tests/validation/json-schemas/protocols/protocols-configure.spec.ts [hunk 2]
```diff
@@ -44,48 +44,46 @@ describe('ProtocolsConfigure schema definition', () => {
 
     expect(() => {
       Message.validateJsonSchema(message);
-    }).throws('who: must be equal to one of the allowed values');
+    }).throws('/$actions/0'); //
```
Comment (thehenrytsai) [inline tests/validation/json-schemas/protocols/protocols-configure.spec.ts:47]: Nit:
```suggestion
    }).throws('/$actions/0');
```

**Review by thehenrytsai — DISMISSED** [review]

File: json-schemas/protocol-rule-set.json (hunk 1 above)
Comment (diehuxx) [reply json-schemas/protocol-rule-set.json:23]: Added

**Review by thehenrytsai — APPROVED** [review]
🙏

**Pull request merged**